### PR TITLE
final addstock functionality

### PR DESCRIPTION
**Pull request by princekumarLnG**
Added full functionality for saving stocks to the database.

### CODE REVIEW

File: Pspcl/Pspcl.Web/Controllers/StockViewController.cs [hunk 1]
```diff
@@ -62,53 +62,88 @@ public JsonResult getMaterialCodes(int materialTypeId)
...
+			{
+				var model = new StockViewModel();
+				//var formValues = HttpContext.Request.Form;
+				//// Convert the FormCollection to a dictionary
+				//var formData = formValues.ToDictionary(x => x.Key, x => x.Value.ToString());
+				//// Store the serialized dictionary in the temp data
+				//TempData["FormData"] = JsonConvert.SerializeObject(formData);
+
+				model.SelectedMaterialCode = formCollection["selectedMaterialCode"];
+
```
Comment (VikasNischalDA) [inline Pspcl/Pspcl.Web/Controllers/StockViewController.cs:77]: Why we assigning each value here? why not using automapper

File: Pspcl/Pspcl.Web/Controllers/StockViewController.cs [hunk 2]
```diff
@@ -62,53 +62,88 @@ public JsonResult getMaterialCodes(int materialTypeId)
...
+						SerialNumberFrom = Convert.ToInt32(element_from.Value),
+						SerialNumberTo = int.Parse(element_to.Value),
+						Quantity = int.Parse(element_qty.Value),
+					};
+					stockMaterialsList.Add(row);
+				}
+
+				model.stockMaterialList = stockMaterialsList;
+
+				if (ModelState.IsValid)
```
Comment (VikasNischalDA) [inline Pspcl/Pspcl.Web/Controllers/StockViewController.cs:112]: why we checking model state at start of the project

File: Pspcl/Pspcl.Web/Controllers/StockViewController.cs (hunk 1 above)
Comment (sachreetkaur) [reply Pspcl/Pspcl.Web/Controllers/StockViewController.cs:77]: actually vikas these things are done by simran and we use form collection that have data in the form of  key value pair that is coming from razor view and we have fetch it to fill our model object . We are mapping data here in the controller.

**Review by VikasNischalDA — CHANGES_REQUESTED** [review]
Added comments

File: Pspcl/Pspcl.Web/Controllers/StockViewController.cs (hunk 2 above)
Comment (sachreetkaur) [reply Pspcl/Pspcl.Web/Controllers/StockViewController.cs:112]: when the model is ready and before saving it into database.,we have to check whether the model state is valid like all the required fields are filled or not or it is of correct data-type.

**Review by VikasNischalDA — APPROVED** [review]

**Pull request merged**